code/client/client.cpp

```diff
@@ -0,0 +1,61 @@
+#include <iostream>
+#include <yarp/os/all.h>
+
+using namespace std;
+using namespace yarp::os;
+
+int main(int argc, char * argv[])
+{
+	if(argc<2){
+		cout << "Please supply a port name for client" << endl;
+		return 1;
+	}
+
+	const char *charName = argv[1];
+
+    Network yarp;	//initialize yarp
+    Semaphore sem;
+    Port portP;
+    portP.open(charName);
+    
+	int intN = 0, intTh = 0;
+
+	while(true){
+		if(portP.getOutputCount() == 0){
+			cout << "Trying to connect to server" << endl;
+			yarp.connect(charName, "/server");
+		}else{
+		    Bottle botRequest;
+		    Bottle botResponse;
+
+		    //make request and send it
+			botRequest.addString("vocab_req");
+			botRequest.addInt(intN);
+			portP.write(botRequest, botResponse);	//request and wait response
+
```

Comment:
Oh, I didn't notice this hpp file.
I fixed to refer this hpp file and to use its constants.